pkg/provisioning/config/yaml/linter.go

```diff
@@ -49,16 +57,23 @@ func (cl *configLinter) init() {
 func (cl *configLinter) InspectNode(path []string, node *yaml.Node) {
 	if len(path) == 1 && path[0] == "version" {
 		// version gets special treatment, it adjusts the warning we create
-		if _, ok := expandedChangelog[node.Value]; !ok {
+		if _, ok := cl.expandedChangelog[node.Value]; !ok {
 			cl.addWarning(path[0], node, fmt.Sprintf("unrecognized version, using parser version %v instead", cl.version))
 			return
 		}
 		cl.version = node.Value
 		return
 	}
 
-	curMap := expandedChangelog[cl.version]
-	for _, field := range path {
+	if c, ok := cl.findChange(path); ok {
+		cl.addWarning(path[len(path)-1], node, c.Message)
+	}
+}
+
+func (cl *configLinter) findChange(path []string) (internal.Change, bool) {
```

Comment:
This logic was adjusted a bit so that we don't output all warnings that match anything in `path` but just output the one warning that matches the whole `path`.

For example, if we previously had a change in the changelog tied to the field `foo.bar` and the path contained `[foo,bar,another,field]` we produced a warning for this line because of `foo.bar`. This also meant that we produced a warning for each descendant element that was a leaf (has no children). The reason was that the `yaml` library triggered the hook only for leaf nodes. Now the `yaml` lib triggers the hook for all nodes (including inner nodes), so we can target the warnings better.